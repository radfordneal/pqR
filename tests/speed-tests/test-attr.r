source("time.r")
 
a <- 9
attributes(a) <- list (dd=1.4, cc=1.3, bb=1.2, aa=1.1)

test.name <- "attr"

n <- 2000000

f <- test.cmp (function () {for (i in 1:n) r <- attr(a,"bb"); r })

sys.time(r<-f())
print(r)

f <- test.cmp (function () {for (i in 1:n) r <- attr(a,"b"); r })

sys.time(r<-f())
print(r)

n <- 1000000

f <- test.cmp (function () {for (i in 1:n) { b <- a; attr(b,"bb") <- NULL }; b})

sys.time(r<-f())
print(r)

n <- 1000000

f <- test.cmp (function () {for (i in 1:n) { b <- a; attr(b,"bb") <- "x" }; b})

sys.time(r<-f())
print(r)
