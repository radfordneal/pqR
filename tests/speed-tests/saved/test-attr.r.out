
---- attr 

Time:
[1] 1.2
Time:
[1] 1.2
Time:
[1] 9
attr(,"dd")
[1] 1.4
attr(,"cc")
[1] 1.3
attr(,"aa")
[1] 1.1
Time:
[1] 9
attr(,"dd")
[1] 1.4
attr(,"cc")
[1] 1.3
attr(,"bb")
[1] "x"
attr(,"aa")
[1] 1.1
